### PR TITLE
Consistency in docstring format and naming of variables

### PR DESCRIPTION
**Pull request by fhagemann**
In order to make consistent docstrings for `structs`, `functions` and `dictionaries`, docstring templates were created.

Descriptions should begin with an upper-case letter and end with a period.
Sections should be defined with two hashtags `##`.
Avoid multiple docstrings for similar functions (see `Grid(::Simulation)` in 9353742, try to unify them)

For `structs`:
Short description, (if relevant) parametric types and fields.

```julia
"""
    struct StructName{T, N, S, AT} <: AbstractArray{T, N}
        
Description.
        
## Parametric types (only if relevant, no need if it's only `{T}`)
* `T`: Description.
* `N`: Description.
* `S`: Description.
* `AT`: Description.
        
## Fields
* `data::Array{T, N}`: Description.
* `grid::Grid{T, N, S, AT}`: Description.
"""
struct StructName{T, N, S, AT} <: AbstractArray{T, N}
    data::Array{T, N}
    grid::Grid{T, N, S, AT}
end
```

For `functions`:
Short description, arguments, (if applicable) keyword arguments, example code and (if needed) notes)
```julia
"""
    function_name(arguments::A; keyword::K)::Nothing

Description (and optionally, what it returns)

## Arguments
* `arguments::A`: Description.

## Keywords
* `keyword::K`: Description.

## Examples
    function_name(example_argument, example_keyword)

!!! note 
    Something that should be kept in mind. (e.g. calculate `electric_potential` before `electric_field`)
"""
function function_name(arguments::A; keyword::K)::Nothing
    # ...
end
```

The docstrings for `structs` should have already been updated, docstrings for `functions` are still missing.

### CODE REVIEW

**Comment by fhagemann** [comment]
Docstrings still need to be made consistent for:
- [ ] functions (except `calculate_electric_potential!`, `calculate_electric_field!`, `calculate_drift_fields!`)
- [ ] CSG

**Comment by fhagemann** [comment]
All existing CSG docstrings were updated.

**Review by fhagemann — COMMENTED** [review]
I adjusted the names of variables such that the instance of a specific type ideally always has the same name
(`sim` for Simulation, `det` for SolidStateDetector, `grid` for Grid, `evt` for Event, `ef` for ElectricField, `epot` for ElectricPotential, `wpot` for WeightingPotential, `point_types` for PointTypes, `pt` for Points). However, I might have missed some.

The renaming comes with a change in the struct `PotentialSimulationSetup`. Will that have any serious consequences for users?

File: src/PotentialSimulation/PotentialSimulationSetups/PotentialSimulationSetups.jl [hunk 1]
```diff
@@ -8,7 +8,7 @@ abstract type AbstractPotentialSimulationSetup{T, N} end
 struct PotentialSimulationSetup{T, N, S} <: AbstractPotentialSimulationSetup{T, N}
     grid::Grid{T, N, S}
     potential::Array{T, N}
-    pointtypes::Array{PointType, N}
+    point_types::Array{PointType, N}
```
[the remark this answers is not in the document]
Comment (lmh91) [reply src/PotentialSimulation/PotentialSimulationSetups/PotentialSimulationSetups.jl:11]: No, its fine.

**Comment by fhagemann** [comment]
I think that all docstrings are updated now.

However, building the docs locally takes forever and I do not know whether that is just an issue on my laptop or something realted to the docstrings...

File: src/PlotRecipes/Potentials.jl [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-        return get_crosssection_idx_and_value(g, r, T(0.0), z)
+        return get_crosssection_idx_and_value(grid, r, T(0.0), z)
     elseif !ismissing(φ) && ismissing(r) && ismissing(z)
         φ_rad::T = T(deg2rad(φ))
-        while !(g.φ.interval.left <= φ_rad <= g.φ.interval.right) && g.φ.interval.right != g.φ.interval.left
-            if φ_rad > g.φ.interval.right
-                φ_rad -= width(g.φ.interval)
-            elseif φ_rad < g.φ.interval.left
-                φ_rad += width(g.φ.interval)
+        while !(grid.φ.interval.left <= φ_rad in grid.φ.interval.right) && grid.φ.interval.right != grid.φ.interval.left
```
Comment (fhagemann) [inline src/PlotRecipes/Potentials.jl]: This commit (edd265f) caused the bug that prevented the docs from building.
It is now hopefully fixed with 7ac770f.

**Comment by fhagemann** [comment]
Things that have changed other than docstrings:
- The folders of charge drift models were renamed (f282cb9, 4c13fc6, 53db103)
- Capacitance code is moved to `Simulation/Capacitance.jl` (18cec45)
- Output is generated when a config file with old format is read in (a726154)

**Pull request merged**